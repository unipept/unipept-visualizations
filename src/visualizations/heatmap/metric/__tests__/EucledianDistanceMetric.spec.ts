import TestDataGenerator from "./../../../../test/TestDataGenerator";
import EuclidianDistanceMetric from "./../EuclidianDistanceMetric";

describe("EuclidianDistanceMetric", () => {
    it('should produce lower triangular matrices', () => {
        const dataGenerator: TestDataGenerator = new TestDataGenerator();
        const data: number[][] = dataGenerator.generateSmall2DDataset();

        const euclideanMetric: EuclidianDistanceMetric = new EuclidianDistanceMetric();
        const distanceMatrix: number[][] = euclideanMetric.getDistance(data);

        for (let i: number = 0; i < distanceMatrix.length; i++) {
            expect(distanceMatrix[i].length).toBe(i + 1);
        }
    });

    it('should correctly calculate euclidean distances for small 2D dataset', () => {
        const dataGenerator: TestDataGenerator = new TestDataGenerator();
        const data: number[][] = dataGenerator.generateSmall2DDataset();

        const euclideanMetric: EuclidianDistanceMetric = new EuclidianDistanceMetric();
        const distanceMatrix: number[][] = euclideanMetric.getDistance(data);

        expect(distanceMatrix).toMatchSnapshot();
    });
});
