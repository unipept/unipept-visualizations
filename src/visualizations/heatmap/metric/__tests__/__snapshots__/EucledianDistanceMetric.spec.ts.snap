// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EuclidianDistanceMetric > should correctly calculate euclidean distances for small 2D dataset 1`] = `
[
  [
    0,
  ],
  [
    0.19999999999999996,
    0,
  ],
  [
    1.5206906325745548,
    1.5660459763365826,
    0,
  ],
  [
    2.23606797749979,
    2.154065922853802,
    1.346291201783626,
    0,
  ],
  [
    2.5,
    2.3853720883753127,
    1.8200274723201295,
    0.5,
    0,
  ],
]
`;
